src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -844,11 +853,134 @@ private static string MakeShortCategoryString(string fullCategoryName)
         #region Selection
 
         /// <summary>
-        ///     Increments the selected element by 1, unless it is the last element already
+        /// Selected member is  library search view.
+        /// </summary>
+        public NodeSearchElementViewModel CurrentlySelectedMember
+        {
+            get
+            {
+                var selectedCategory = SearchRootCategories[selectedCategoryIndex];
+                var selectedMemberGroup = selectedCategory.MemberGroups.ElementAt(selectedMemberGroupIndex);
+                var selectedMember = selectedMemberGroup.Members.ElementAt(SelectedMemberIndex);
+
+                return selectedMember;
```

Comment:
Instead of having a separate `selectedMemberGroupIndex` and `SelectedMemberIndex` to manage selection, can we not search for the selected member here in this getter? Having two more variables to manage the same thing it already manages internally (with `CurrentlySelectedMember.IsSelected` property) is duplicating code and making things more error-prone (imagine what happens when these variables go out-of-sync).

If this property is not called in high frequency and the data set is not too huge, then I'll suggest to go without these data members -- if we need to find the selected member, traverse the tree to find it.

Does that make sense?